src/pybind/mgr/dashboard/services/rgw_client.py

```diff
@@ -796,6 +797,45 @@ def list_roles(self) -> List[Dict[str, Any]]:
 
         return roles
 
+    def create_role(self, role_name: str, role_path: str, role_assume_policy_doc: str) -> None:
+        try:
+            json.dumps(json.loads(role_assume_policy_doc))
+        except:
+            raise DashboardException(f'Assume role policy doc is not a valid json') 
+        def between(v, a, b):
+            return ord(v) >= ord(a) and ord(v) <= ord(b)
+        valid_special = '_+=,.@'
+        if len(role_name) > 64:
+            raise DashboardException(f'Role name "{role_name}" is invalid. Should be 64 characters or less') 
+        for c in role_name:
+            if not (between(c, 'a', 'z') or between(c, 'A', 'Z') or c in valid_special):
+                raise DashboardException(f'Role name "{role_name}" is invalid. Valid characters are [a-zA-Z_+=,.@]') 
+
+        if not os.path.isabs(role_path):
+            raise DashboardException(f'Role path "{role_path}" is invalid. It should be an absolute path') 
+        if role_path[-1] != '/':
+            raise DashboardException(f'Role path "{role_path}" is invalid. It should start and end with a slash') 
+        for c in role_path:
+            if ord(c) > 0x7F or ord(c) < 0x21:
```

Comment:
@epuertat pinging you because I know you'll love to review this in the future :)